vulture/core.py

```diff
@@ -489,40 +489,35 @@ def generic_visit(self, node):
 
 
 def _parse_args():
-    def csv(option, _, value, parser):
-        setattr(parser.values, option.dest, value.split(','))
-    usage = """\
-usage: %prog [options] PATH [PATH ...]
-
-Paths may be Python files or directories. For each directory vulture
-analyzes all contained *.py files.
-"""
+    usage = "%(prog)s PATH [PATH ...] [options]"
     version = "vulture {0}".format(__version__)
-    parser = optparse.OptionParser(usage=usage, version=version)
-    parser.add_option(
-        '--exclude', action='callback', callback=csv, metavar='PATTERN',
-        type='string', default=[],
-        help=(
-            'Comma-separated list of paths to ignore (e.g.,'
-            ' *settings.py,docs/*.py). PATTERNs can contain globbing'
-            ' characters (*, ?, [, ]). Treat PATTERNs without globbing'
-            ' characters as *PATTERN*.'))
-    parser.add_option(
-        "--sort-by-size", action="store_true",
-        help="Sort unused functions and classes by their lines of code")
-    parser.add_option('-v', '--verbose', action='store_true')
-    parser.add_option(
-        '--min-confidence', action='store', type='int', default=0, help=(
+    parser = argparse.ArgumentParser(prog='vulture', usage=usage)
+    parser.add_argument(
+        'paths', action='store', nargs='*', metavar='PATH', help=(
+            'Paths may be Python files or directories. For each directory'
+            ' vulture analyzes all contained *.py files.'))
+    parser.add_argument(
+        '--exclude', action='store', metavar='PATTERN', type=str, default=[],
+        nargs="+", help=(
+            'Paths to ignore (e.g., *settings.py docs/*.py). PATTERNs can'
```

Comment:
I think that `vulture --exclude *settings.py docs/*.py myprogram.py`  would check (and not ignore) the docs folder.

Reply:
I think we should add tests for this by listing excludes not as `--exclude tests/*`, but `--exclude tests/myfile.py tests/mysecondfile.py` in one test. (Choose files appropriately.)

Reply:
The command `vulture --exclude *settings.py docs/*.py myprogram.py` would just exclude all the three because we are taking up any number of arguments occurring after `--exclude`. This is also the reason why I've placed PATH before [options].

Reply:
> I think we should add tests for this by listing excludes not as --exclude tests/*, but --exclude tests/myfile.py tests/mysecondfile.py in one test. (Choose files appropriately.)

Yes, I see that we aren't explicitly testing `--exclude`. I am gonna add a test.


One more thing I've been wondering: If we use glob patterns, wouldn't shell automatically expand it to the list of all matching files even before passing it on to vulture. This can have unwanted side effects - Let's take up the example `vulture vulture/ --exclude *settings.py` - If no file matches the glob `*settings.py`, shell would automatically abort the process before even starting to execute Vulture. I think the expected behavior should be to silently ignore the glob if no files are found.

Reply:
No, I think we should abort in that case. Vulture just cannot distinguish between a list generated by the user or a shell-glob.